src/client.js

```diff
@@ -157,6 +157,36 @@ export default class Client {
     });
   }
 
+  fetchAllCollectionsWithProducts() {
+    const query = collectionConnectionQuery([...collectionDefaultFields, ['products', productConnectionQuery()]]);
+    const rootQuery = this.graphQLClient.query((root) => {
+      root.add('shop', (shop) => {
+        query(shop, 'collections');
+      });
+    });
+
+    return this.graphQLClient.send(rootQuery).then(({model}) => {
+      const promises = model.shop.collections.reduce((promisesAcc, collection) => {
+        return collection.products.reduce((promiseAcc, product) => {
+          // Fetch the rest of the images and variants for this product
+          promiseAcc.push(this.graphQLClient.fetchAllPages(product.images, {pageSize: 250}).then((images) => {
+            product.attrs.images = images;
+          }));
+
+          promiseAcc.push(this.graphQLClient.fetchAllPages(product.variants, {pageSize: 250}).then((variants) => {
+            product.attrs.variants = variants;
+          }));
+
+          return promiseAcc;
+        }, []);
+      }, []);
```

Comment:
This probably shouldn't fetch all products. Fetch the first set, and we expose pagination on the client, so let people paginate. I'd limit this to something much smaller like 20? But at least make it a param on the method.